app.py

```diff
@@ -136,6 +146,61 @@ def prepare_response_received(random_delay, random_case):
     messages_to_send.append(message)
 
 
+def prepare_response_received_pqrs(random_delay, random_case):
+    prepare_response_received_pqrs_and_qm(random_delay, random_case, channel="PQRS")
+
+
+def prepare_response_received_qm(random_delay, random_case):
+    prepare_response_received_pqrs_and_qm(random_delay, random_case, channel="QM")
+
+
+def prepare_response_received_pqrs_and_qm(random_delay, random_case, channel):
+    tx_id = str(uuid.uuid4())
+
+    message_contents = {
+        "channel": channel,
+        "dateTime": f"{datetime.utcnow().isoformat()}Z",
+        "questionnaireId": random_case['qid'],
+        "transactionId": str(uuid.uuid4()),
+        "type": "RESPONSE_RECEIVED"
+    }
+
+    message = {
+        "type": "PUBSUB",
+        "topic": Config.OFFLINE_RECEIPT_TOPIC_NAME,
+        "project": Config.OFFLINE_RECEIPT_TOPIC_PROJECT_ID,
+        "delay": random_delay,
+        "message_body": json.dumps(message_contents),
+        "tx_id": tx_id
+    }
+
+    messages_to_send.append(message)
+
+
+def prepare_response_received_qm_blanks(random_delay, random_case):
```

Comment:
you could enhance the `prepare_response_received_pqrs_and_qm` function to take an `unreceipt` flag